src/client.ts

```diff
@@ -0,0 +1,84 @@
+import algosdk from 'algosdk'
+import AlgokitComposer, {
+  AppCallParams,
+  AssetConfigParams,
+  AssetCreateParams,
+  AssetDestroyParams,
+  AssetFreezeParams,
+  AssetTransferParams,
+  KeyRegParams,
+  MethodCallParams,
+  PayTxnParams,
+} from './composer'
+
+export default class AlgokitClient {
+  algod: algosdk.Algodv2
+
+  signers: { [address: string]: algosdk.TransactionSigner } = {}
+
+  cachedSuggestedParamsTimeout: number = 3000 // three seconds
+
+  cachedSuggestedParams?: { params: algosdk.SuggestedParams; time: number }
+
+  defaultSigner?: algosdk.TransactionSigner
+
+  constructor({ algodClient, defaultSigner }: { algodClient: algosdk.Algodv2; defaultSigner?: algosdk.TransactionSigner }) {
+    this.algod = algodClient
+    this.defaultSigner = defaultSigner
+  }
+
+  async getSuggestedParams() {
+    if (this.cachedSuggestedParams && Date.now() - this.cachedSuggestedParams.time < this.cachedSuggestedParamsTimeout) {
+      return this.cachedSuggestedParams.params
+    }
+
+    const params = await this.algod.getTransactionParams().do()
+    this.cachedSuggestedParams = { params, time: Date.now() }
+
+    return params
+  }
+
+  newGroup() {
```

Comment:
Should we call this `.compose()` to match the generated typed client?

Reply:
I intentionally went with `newGroup` because it's self descriptive. Anyone reading the code will deduce that executing this function will start a new group. `compose` on the other hand is not immediately obvious unless you are familiar with the existing terminology used by the generated clients and ATC.